pkg/controller/master/upgrade/version_syncer.go

```diff
@@ -226,13 +233,13 @@ func (s *versionSyncer) getNewVersion(v Version) (*harvesterv1.Version, error) {
 	return &newVersion, nil
 }
 
-func canUpgrade(currentVersion string, newVersion *harvesterv1.Version) bool {
+func canUpgrade(currentVersion string, newVersion *harvesterv1.Version, responderVersion Version) bool {
 	switch {
 	case newVersion.Spec.ISOURL == "" || newVersion.Spec.ISOChecksum == "":
```

Comment:
If these two cases of version objects should never exist in the cluster, maybe we can move this to the validator to simplify the logic of the `canUpgrade` function?

Reply:
we could do it later, it seems to have been around forever